test/e2e/installplan_e2e_test.go

```diff
@@ -960,6 +1015,260 @@ func TestInstallPlanWithCRDSchemaChange(t *testing.T) {
 	}
 }
 
+func TestInstallPlanWithDeprecatedVersionCRD(t *testing.T) {
+	defer cleaner.NotifyTestComplete(t, true)
+
+	// generated outside of the test table so that the same naming can be used for both old and new CSVs
+	mainCRDPlural := genName("ins")
+
+	// excluded: new CRD, same version, same schema - won't trigger a CRD update
+	tests := []struct {
+		name            string
+		expectedPhase   v1alpha1.InstallPlanPhase
+		oldCRD          *apiextensions.CustomResourceDefinition
+		intermediateCRD *apiextensions.CustomResourceDefinition
+		newCRD          *apiextensions.CustomResourceDefinition
+	}{
+		{
+			name:          "upgrade CRD with deprecated version",
```

Comment:
Since you only have one scenario, do you think additional tests should be added in the future? If so, how about add them now? Since this e2e is kind of lengthy, I wonder if "smooshing" it all together might make it a little shorter.

Reply:
This is a test case for standard deprecating a CRD version flow that is recommended on the doc. So it is only one case to properly remove a CRD version. I can add a failure case but it is a failure on k8s side as users don't follow the k8s doc. So the new `removeDeprecatedStoredVersions` function won't do anything so the failure will be on the k8s `Update` func. I don't think it is worth a test case to basically test a k8s func. The unit test case is to test the those rejected conditions for `removeDeprecatedStoredVersions`. That is faster and more efficient given the e2e test case is as long as it is, not just code-wise but steps-wise.

Reply:
There are sone duplicated codes so I can group them into some functions to reduce the length.